homeassistant/components/filesize/sensor.py

```diff
@@ -43,10 +44,15 @@ def setup_platform(
     setup_reload_service(hass, DOMAIN, PLATFORMS)
 
     sensors = []
+    pathlist = []
     for path in config[CONF_FILE_PATHS]:
+        if path in pathlist:
+            continue
+        pathlist.append(path)
```

Comment:
They can create multiple instances of the platform so its still possible the same path could be added twice

Reply:
The integration could be changed to the newer format described in ADR0007 which would make it possible to ensure its unique.

https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md

Unfortunately thats a breaking change for a lot of people. 

You could allow both ways to setup for now and warn that the current way is going away and they need to migrate.  Once the migration period is over, the unique id could be added.

Reply:
Might make more sense to convert it to a config flow instead

Reply:
I agree but I don't think that would be allowed (yet)?
I was thinking perhaps to do it once the rules are "softened" but in the meantime just get all other up to code.
Worth to roll in a breaking change now and then another one later (config flow)?

Reply:
If it's a config flow, we should limit the passed in paths to be part of the configured allowed directories (which can only be configured in YAML). I don't know if that would be very useful since yuo already have to edit YAML.

Reply:
I didn't realize there was a limit on allowed directories.   I'm assuming it is some type of anti-foot shooting safety since you don't have to cross a security boundary to configure it?

Reply:
I think what 95% does is to use this to check their db size (my guess) and it's already by default allowed. For the other more advanced users I don't think adding directories to yaml and configure this in UI would be "strange" in any sense.
From a user-friendliness I would bring this into config flow even in the case someone has to modify the allowed path manually.

Reply:
It looks like it was added in #8189. It looks like the original intent was to prevent paths from being accessed from various services but later got extended to configuration as well?

I don't understand the motivation for its use here as its not an effective security measure since same privilege is required is required to change the value as edit `configuation.yaml` or create a new config entry.

Reply:
https://github.com/home-assistant/core/pull/12211#discussion_r167447137

Reply:
Based on my previous comment, I'd say its ok to remove the check, but I don't feel comfortable making that call.

Reply:
As just posted @balloob made the original call for this integration to check whitelist path so I assume we better off waiting his comments before actioning.

Reply:
I agree. I see how this could be useful for services where the input data isn't controlled by the user, but in this case, all the input is from the authenticated trusted user so maybe I'm just missing something here but it seems like all the check does it make if more difficult for the user since they already have all the privileges they need to overcome the check.

Reply:
Yeah, I agree that doing it for config based things is not very useful.  Anything coming in via services / configuration from the UI needs to have these checks.

Reply:
Maybe I did a 👍 to quick here but reading your comment a second time means if we stay in yaml we can remove the check but if we go config flow we need to maintain it?

Reply:
correct